python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1573 @@
+            # Fallback: use exposure center
+            self.log.warning(
+                "Could not get sky positions from donut stamps, using exposure center as fallback. "
+                "This may indicate missing WCS information or stamp metadata."
+            )
+            bbox = exposure.getBBox()
+            centerPos = exposure.wcs.pixelToSky(bbox.getCenterX(), bbox.getCenterY())
+            donutData["coord_ra"] = [centerPos.getRa().asRadians()] * numDonuts
+            donutData["coord_dec"] = [centerPos.getDec().asRadians()] * numDonuts
+
+        # Get centroid positions from stamps metadata (already populated by
+        # createDonutStampsFromTarts)
+        centroidPositions = donutStamps.getCentroidPositions()
+        donutData["centroid_x"] = [pos.getX() for pos in centroidPositions]
+        donutData["centroid_y"] = [pos.getY() for pos in centroidPositions]
+
+        # Get fx, fy, snr from stamps metadata (already populated by
+        # createDonutStampsFromTarts)
+        donutData["fx"] = self._getMetadataArray(donutStamps.metadata, "FX", numDonuts, 0.0)
+        donutData["fy"] = self._getMetadataArray(donutStamps.metadata, "FY", numDonuts, 0.0)
+        donutData["snr"] = self._getMetadataArray(donutStamps.metadata, "SN", numDonuts, np.nan)
+
+        # Add detector information as a column (required by downstream
+        # aggregation)
+        # Ensure detector name is consistent with expected format
+        donutData["detector"] = [detectorName] * numDonuts
+        self.log.info(
+            "Using detector name: '%s' for %d donuts (defocal type: '%s')",
+            detectorName,
+            numDonuts,
+            defocalType,
+        )
+
+        # Add missing telescope coordinate columns required by downstream
+        # aggregation
+        # These are not generated by neural network, so use default values
+        donutData["thx_ccs"] = [np.nan] * numDonuts  # Telescope X in Camera Coordinate System
+        donutData["thy_ccs"] = [np.nan] * numDonuts  # Telescope Y in Camera Coordinate System
+        donutData["thx_ocs"] = [np.nan] * numDonuts  # Telescope X in Observatory Coordinate System
+        donutData["thy_ocs"] = [np.nan] * numDonuts  # Telescope Y in Observatory Coordinate System
+        donutData["th_n"] = [np.nan] * numDonuts  # Telescope North coordinate
+        donutData["th_w"] = [np.nan] * numDonuts  # Telescope West coordinate
+
+        # Create the table
+        donutTable = QTable(donutData)
+
+        # Add units matching standard format
+        donutTable["coord_ra"].unit = u.rad  # Use radians like standard
+        donutTable["coord_dec"].unit = u.rad  # Use radians like standard
+        # Note: centroidX and centroidY are kept as plain floats to avoid
+        # Point2D constructor issues
+        donutTable["fx"].unit = u.deg
+        donutTable["fy"].unit = u.deg
+
+        # Add comprehensive visitInfo metadata matching standard format
+        try:
```

Comment:
Isn't this code all exactly what this function does?

https://github.com/lsst-ts/ts_wep/blob/982c2eab0d22739dd2bd7603ace8b1f224b2dffd/python/lsst/ts/wep/task/generateDonutCatalogUtils.py#L215

As I mentioned above, let's adjust that code if we need to so we're not replicating large blocks of essentially the same functionality.